src/Apps/Order/Routes/__tests__/Payment.test.tsx

```diff
@@ -0,0 +1,197 @@
+import { mount, shallow } from "enzyme"
+import React from "react"
+
+import { Checkbox, CheckboxProps } from "@artsy/palette"
+import { OrderWithShippingDetails } from "Apps/__test__/Fixtures/Order"
+import { AddressForm } from "../../Components/AddressForm"
+import { ContinueButton, PaymentProps, PaymentRoute } from "../Payment"
+
+jest.mock("react-relay", () => ({
+  commitMutation: jest.fn(),
+  createFragmentContainer: component => component,
+}))
+
+jest.mock("react-stripe-elements", () => ({
+  CardElement: props => <div {...props} />,
+  injectStripe: args => args,
+}))
+
+import { Sans } from "@artsy/palette"
+import { commitMutation, RelayProp } from "react-relay"
+import { creatingCreditCardSuccess } from "../../../__test__/Fixtures/MutationResults/createCreditCard"
+import { settingOrderPaymentSuccess } from "../../../__test__/Fixtures/MutationResults/setOrderPayment"
+import {
+  CreditCardErrorMessage,
+  CreditCardInput,
+} from "../../Components/CreditCardInput"
+import { TwoColumnLayout } from "../../Components/TwoColumnLayout"
+
+const mutationMock = commitMutation as jest.Mock<any>
+
+describe("Payment", () => {
+  let stripeMock
+  let testProps: PaymentProps
+
+  beforeEach(() => {
+    mutationMock.mockReset()
+
+    stripeMock = {
+      createToken: jest.fn(),
+    }
+
+    testProps = {
+      order: { ...OrderWithShippingDetails, id: "1234" },
+      relay: { environment: {} } as RelayProp,
+      router: { push: jest.fn() },
+      stripe: stripeMock,
+    } as any
+  })
+
+  it("tokenizes credit card information using shipping address as billing address", () => {
+    const thenMock = jest.fn()
+    stripeMock.createToken.mockReturnValue({ then: thenMock })
+
+    mount(<PaymentRoute {...testProps} />)
+      .find(ContinueButton)
+      .simulate("click")
+
+    expect(stripeMock.createToken).toHaveBeenCalledWith({
+      name: "Joelle Van Dyne",
+      address_line1: "401 Broadway",
+      address_line2: "Suite 25",
+      address_city: "New York",
+      address_state: "NY",
+      address_zip: "10013",
+      address_country: "US",
+    })
+    expect(thenMock.mock.calls.length).toBe(1)
+  })
+
+  it("tokenizes credit card information with a different billing address", () => {
+    const thenMock = jest.fn()
+    stripeMock.createToken.mockReturnValue({ then: thenMock })
+
+    const paymentRoute = mount(<PaymentRoute {...testProps} />)(
+      paymentRoute.find(Checkbox).props() as CheckboxProps
+    ).onSelect(false)
```

Comment:
I was pairing w/ @yuki24 on this... for whatever reason, `simulate('select', false)` wasn't working here. I've seen a few conversations about similar things online, but we weren't really able to figure out why it wasn't working. Does anyone else have any ideas?

Reply:
Yeah. `simulate` behaves differently depending on whether you do shallow mount or regular old mount.

- With shallow mount, simulate just calls the prop that maps to the event name you provide. So `blah.simulate('foo', data)` calls `blah.props().onFoo(data)`.
- With regular old mount, simulate simulates actual dom events. It will find the outermost actual dom rendered by the component you call it on, then trigger an event on that node, merging whatever data you provide with a fully-fleshed out event object. So `blah.simulate('foo', data)` is something like `ReactDOM.findDomNode(blah).trigger('foo', Object.assign(new WhateverEvent(), data))`

so to fix this you need to narrow your selection to the actual dom node that react attaches the event handler to, or just use the props like you already did. A lot of people do the latter 🤷‍♂️

Reply:
Cool! Thanks for the information!

I guess we can leave it as is then.